### PR TITLE
style(xy): expose isolated point style

### PR DESCRIPTION
**Pull request by markov00**
## Summary

Fix part of #1902 by exposing the style of the isolated points.

Now the previously called `orphan` data points are renamed to `isolated` points to follow a more [mathematical semantic](https://en.wikipedia.org/wiki/Isolated_point).

Screenshot 2023-03-27 at 15 14 06

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [ ] The `:theme` label has been added and the `@elastic/eui-design` team has been pinged when there are `Theme` API changes
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] New public API exports have been added to `packages/charts/src/index.ts`
- [x] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated
- [x] The code has been checked for cross-browser compatibility (Chrome, Firefox, Safari, Edge)
- [x] Visual changes have been tested with all available themes including `dark`, `light`, `eui-dark` & `eui-light`

### CODE REVIEW

**Comment by markov00** [comment]
buildkite update screenshots

**Review by nickofthyme — APPROVED** [review]
Not sure if you were waiting on my review or not, but the changes look good to me.

**Comment by markov00** [comment]
> Not sure if you were waiting on my review or not, but the changes look good to me.

Thanks, no I'm just waiting to test this in Kibana directly. I probably will consider exposing this configuration in the theme and tweaking this in Kibana rather than having this hardcoded

File: packages/charts/src/chart_types/xy_chart/rendering/points.ts [hunk 1]
```diff
@@ -81,17 +82,22 @@ export function renderPoints(
       const seriesIdentifier: XYChartSeriesIdentifier = getSeriesIdentifierFromDataSeries(dataSeries);
       const styleOverrides = getPointStyleOverrides(datum, seriesIdentifier, styleAccessor);
       const style = buildPointGeometryStyles(color, pointStyle, styleOverrides);
-      const orphan = isOrphanDataPoint(dataIndex, dataSeries.data.length, yDefined, prev, next);
+      const isPointIsolated =
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/xy_chart/rendering/points.ts:85]: Technically this is more like should we apply isolated point styles, not if the point is isolated. Change if you like, fine as is.

File: packages/charts/src/chart_types/xy_chart/rendering/points.ts [hunk 2]
```diff
@@ -105,7 +111,7 @@ export function renderPoints(
         },
         seriesIdentifier,
         panel,
-        orphan,
+        isolated: isPointIsolated,
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/xy_chart/rendering/points.ts:114]: Do we care that this depends on `isolatedPointThemeStyle.visible` being `true`/`false`? If this is accessible to the consumer I think we change this to...

```suggestion
        isolated: isIsolatedPoint(dataIndex, dataSeries.data.length, yDefined, prev, next),
```

File: packages/charts/src/chart_types/xy_chart/rendering/bubble.ts [hunk 3]
```diff
@@ -43,6 +43,8 @@ export function renderBubble(
     panel,
     color,
     seriesStyle.point,
+    // there is no concept of isolated point in bubble chart, so we can just disable it through visibility
+    { ...seriesStyle.point, visible: false },
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/xy_chart/rendering/bubble.ts:47]: 👍🏼

**Review by nickofthyme — COMMENTED** [review]
Changes LGTM, tested locally and see no issues other than the one below. I left a few minor code comments in addition.

The [point shapes story](https://github.com/elastic/elastic-charts/pull/2004/files?short_path=9847c8c#diff-8319a2d84bed97d09eb8032312aaa75469fd8b01f89e3e18675df3017535ed58) has a regression in the legend symbols.

Screen Recording 2023-05-25 at 04 57 52 PM

File: packages/charts/src/chart_types/xy_chart/rendering/points.ts (hunk 2 above)
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/rendering/points.ts:114]: fixed through  a435a01 and 20868e0

File: packages/charts/src/chart_types/xy_chart/rendering/points.ts (hunk 1 above)
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/rendering/points.ts]: fixed through  a435a01 and 20868e0

**Comment by markov00** [comment]
> Changes LGTM, tested locally and see no issues other than the one below. I left a few minor code comments in addition.
> 
> The [point shapes story](https://github.com/elastic/elastic-charts/pull/2004/files?short_path=9847c8c#diff-8319a2d84bed97d09eb8032312aaa75469fd8b01f89e3e18675df3017535ed58) has a regression in the legend symbols.
> 
> Screen Recording 2023-05-25 at 04 57 52 PM

Thank you @nickofthyme , this was actually a bug not previously discovered. It is fixed in 2199a2e

**Comment by markov00** [comment]
buildkite update screenshots

**Review by nickofthyme — APPROVED** [review]
Thanks, changes LGTM! 🚢

**Pull request merged**